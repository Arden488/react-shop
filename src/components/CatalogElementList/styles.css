.product-list {
  display: grid;
  grid-template-columns: repeat(3, minmax(240px, 1fr));
  grid-gap: 10px;
}