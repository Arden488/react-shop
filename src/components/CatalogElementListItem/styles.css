.product-item:hover {
  box-shadow: 0 0 10px rgba(0, 0, 0, .15);
}

.product-item__image {
  text-align: center;
}

.product-item__image img {
  max-width: 100%;
}

.product-item__link {
  display: block;
  padding: 20px;
  color: black;
  text-decoration: none;
}

.product-item__title {
  display: block;
  margin: 10px 0;
}

.product-item__controls {
  display: grid;
  grid-auto-flow: column;
  justify-content: space-between;
}
