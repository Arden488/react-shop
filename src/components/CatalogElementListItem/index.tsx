import * as React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

interface ICatalogElementListItemProps {
  id: number,
  key: number,
  title: string,
  price: string,
  image: string,
}

const CatalogElementListItem: React.SFC<ICatalogElementListItemProps> = props => (
  <div className="product-item">
    <Link className="product-item__link" to={`/product/${props.id}`}>
      <div className="product-item__image"><img src={`/img/${props.image}`} alt={props.title} /></div>
      <div className="product-item__title">{props.title}</div>
      <div className="product-item__controls">
        <div className="product-item__price">{props.price} ₽</div>
        <div className="product-item__buy">
          <button>Купить</button>
        </div>
      </div>
    </Link>
  </div>
);

export default CatalogElementListItem;
