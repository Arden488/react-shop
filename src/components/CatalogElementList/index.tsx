import * as React from 'react';

import CatalogElementListItem from '../CatalogElementListItem';

import './styles.css';

interface ICatalogElementListProps {
  items: number[]
}

class CatalogElementList extends React.Component<ICatalogElementListProps> {
  public render() {
    const items = this.props.items.map(item => this.renderListItem(item));

    return (
      <div className="product-list">
        {items}
      </div>
    );
  }
  
  private renderListItem(item: number) {
    return <CatalogElementListItem code="1" key={item} />;
  }
}

export default CatalogElementList;
