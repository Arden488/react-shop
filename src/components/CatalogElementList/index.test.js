/* global it, expect, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';

import CatalogElementList from './index';
import CatalogElementListItem from '../CatalogElementListItem';

let component = null;

beforeEach(() => {
  component = shallow(<CatalogElementList />);
});

it('renders without crashing', () => {
  expect(component.exists()).toEqual(true);
});

it('should contain CatalogElementListItem', () => {
  expect(component.find(CatalogElementListItem).length).toBeGreaterThanOrEqual(1);
});
