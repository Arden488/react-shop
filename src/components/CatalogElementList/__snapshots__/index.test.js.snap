// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-list"
>
  <CatalogElementListItem
    code="1"
    title="Товар 1"
  />
  <CatalogElementListItem
    code="2"
    title="Товар 2"
  />
  <CatalogElementListItem
    code="3"
    title="Товар 3"
  />
  <CatalogElementListItem
    code="4"
    title="Товар 4"
  />
  <CatalogElementListItem
    code="5"
    title="Товар 5"
  />
</div>
`;
